mlx/directives/attribute_link_directive.py

```diff
@@ -0,0 +1,71 @@
+from docutils.parsers.rst import directives
+
+from mlx.traceability_exception import TraceabilityException, report_warning
+from mlx.traceable_base_directive import TraceableBaseDirective
+from mlx.traceable_base_node import TraceableBaseNode
+from mlx.traceable_item import TraceableItem
+
+
+class AttributeLink(TraceableBaseNode):
+    """Node that adds one or more attributes to one or more items."""
+
+    def perform_replacement(self, app, collection):
+        """ Processes the attribute-link items.
+
+        The AttributeLink node has no final representation, so it is removed from the tree.
+
+        Args:
+            app: Sphinx application object to use.
+            collection (TraceableCollection): Collection for which to generate the nodes.
+        """
+        filtered_item_ids = collection.get_items(self['filter'])
+        for attribute, value in self['attributes'].items():
+            for item_id in filtered_item_ids:
+                item = collection.get_item(item_id)
+                try:
+                    item.add_attribute(attribute, value)
+                except TraceabilityException as err:
+                    report_warning(err, self['document'], self['line'])
+        self.replace_self([])
+
+
+class AttributeLinkDirective(TraceableBaseDirective):
+    """ Directive to add attributes to items outside of the items' definition.
+
+    The node will be responsible for applying the configuration to the Item. First, all directives must be parsed.
+
+    Syntax::
+
+      .. attribute-sort::
+         :filter: regex
```

Comment:
This implies that "class" and "filter" cannot be used as valid attributes, correct? Are there already constraints preventing us from doing this, or are we introducing them here?

Reply:
Good question. There are already constraints in place. This `:<<attribute>>:` syntax is not new. A warning is reported when the user has defined an attribute that exists in the `option_spec` of a directive. See [here](https://github.com/melexis/sphinx-traceability-extension/blob/master/mlx/traceability.py#L272-283). Such an attribute cannot be used in `:<<attribute>>:` when the attribute name exists in the `option_spec` of the **relevant** directive.

*I removed the `class` option as it is useless for this directive because it doesn't get rendered.*